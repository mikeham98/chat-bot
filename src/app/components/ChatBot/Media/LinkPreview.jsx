import React from 'react';
import styles from '../../../../themes/components/media.scss';
import PropTypes from "prop-types";

const LinkPreview = ({src, image, title, description}) => {
    return (
        <div className={styles.mediaLinkPreviewWrapper} onClick={() => window.open(src)}>
            <img src={image}/>
            <div className={styles.mediaLinkPreviewMeta}>
                <span className={styles.mediaLinkPreviewTitle}>{title}</span>
                <span className={styles.mediaLinkPreviewDescription}>{description}</span>
            </div>
        </div>
    );
};

LinkPreview.propTypes = {
    src: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
};

export default React.memo(LinkPreview);