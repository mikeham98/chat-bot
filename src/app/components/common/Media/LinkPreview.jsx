import React from 'react';
import styles from '../../../../themes/components/media.scss';

const LinkPreview = ({src, image, title, description}) => {
    return (
        <div className={styles.mediaLinkPreviewWrapper} onClick={() => window.open(src)}>
            <img src={image}/>
            <div className={styles.mediaLinkPreviewMeta}>
                <span className={styles.mediaLinkPreviewTitle}>{title}</span>
                <span className={styles.mediaLinkPreviewDescription}>{description}</span>
            </div>
        </div>
    );
};

export default React.memo(LinkPreview);