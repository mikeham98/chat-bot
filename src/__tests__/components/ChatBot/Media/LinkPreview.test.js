import React from 'react';
import {shallow} from 'enzyme';
import LinkPreview from '../../../../app/components/ChatBot/Media/LinkPreview';

describe('LinkPreview', () => {

    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<LinkPreview {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            src: 'http://www.link.co.uk/article',
            title: 'An interesting Article',
            description: 'This article is very interesting...',
            image: 'http://www.link.co.uk/article.jpg',
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return an img tag and render a title and description', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('window.open', () => {
            it('should be called on click of div with the link to the article', () => {
                window.open = jest.fn();
                expect(window.open).not.toHaveBeenCalled();
                wrapper().find('div').first().props().onClick();
                expect(window.open).toHaveBeenCalledTimes(1);
                expect(window.open).toHaveBeenCalledWith('http://www.link.co.uk/article');
            });
        });
    });
});