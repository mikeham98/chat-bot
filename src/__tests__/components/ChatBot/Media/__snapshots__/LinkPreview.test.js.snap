// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkPreview render should return an img tag and render a title and description 1`] = `
<div
  className="mediaLinkPreviewWrapper"
  onClick={[Function]}
>
  <img
    src="http://www.link.co.uk/article.jpg"
  />
  <div
    className="mediaLinkPreviewMeta"
  >
    <span
      className="mediaLinkPreviewTitle"
    >
      An interesting Article
    </span>
    <span
      className="mediaLinkPreviewDescription"
    >
      This article is very interesting...
    </span>
  </div>
</div>
`;
